orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueShovel.kt

```diff
@@ -34,32 +40,48 @@ class QueueShovel(
   private val queue: Queue,
   private val previousQueue: Queue,
   private val registry: Registry,
-  private val activator: Activator
+  private val activator: Activator,
+  private val config: DynamicConfigService
 ) {
   private val log = LoggerFactory.getLogger(javaClass)
 
   private val pollOpsRateId = registry.createId("orca.nu.shovel.pollOpsRate")
   private val shoveledMessageId = registry.createId("orca.nu.shovel.pushedMessageRate")
   private val shovelErrorId = registry.createId("orca.nu.shovel.pushedMessageErrorRate")
 
-  @Scheduled(fixedDelayString = "\${queue.shovel.poll-frequency.ms:50}")
+  @Scheduled(fixedRate = rateMs)
+  fun migrateIfActive() {
+    if (!isActive()) {
+      return
+    }
+
+    log.info("Actively shoveling from $previousQueue to $queue")
```

Comment:
what does the `toString` here print for the `queue` objects, is it useful?

Reply:
at the moment it's not terribly useful (`shoveling from SqlQueue@X to SqlQueue@Y`), but it has the potential to be a bit more useful. I was thinking we could have SqlQueue print the pool name.

Reply:
now useful with keiko 3.6.1 :)